usaspending_api/awards/v2/filters/location_filter_geocode.py

```diff
@@ -108,20 +128,71 @@ def get_fields_list(scope, field_value):
+def return_query_string(use_matview: bool) -> tuple:
+    # Returns query strings based upon if the queryset is for a normalized or de-normalized model
 
-    if use_matview:
+    if use_matview:  # de-normalized
         # Queries going through the references_location table will not require a join
         # Example "pop__county_code"
         q_str = '{0}_{1}'
-        # Matviews use country_code ex: pop_country_code
-        country_code_col = 'country_code'
+        country_code_col = 'country_code'  # Matviews use country_code ex: pop_country_code
     else:
-        # Queries going through the references_location table will a join in the filter
+        # Queries going through the references_location table will require a join in the filter
         # Example "place_of_performance__county_code"
         q_str = '{0}__{1}'
-        # References_location table uses the col location_country_code
-        country_code_col = 'location_country_code'
+        country_code_col = 'location_country_code'  # References_location table uses the col location_country_code
 
     return q_str, country_code_col
+
+
+def create_city_name_queryset(scope: str, list_of_cities: list, country_code: str, state_code: Optional[str] = None):
+    """
+        Given a list of city names and the scope, return a django queryset.
+        scope = "pop" or "recipient_location"
+        list_of_city_names is a list of strings
+        country_code is the country code to limit the query and results from Elasticsearch
+        state_code (optional) is the state code if the search should be limited to that state
+    """
+    matching_awards = set(
+        chain(*[get_award_ids_by_city(scope, city, country_code, state_code) for city in list_of_cities])
+    )
+    result_queryset = Q(pk=None)  # If there are no city results in Elasticsearch, use this always falsey Q filter
+
+    if matching_awards:
+        result_queryset = Q(**{"award_id" + "__in": matching_awards})
+    return result_queryset
+
+
+def get_award_ids_by_city(scope: str, city: str, country_code: str, state_code: Optional[str] = None) -> list:
+    """
+    """
+    # Search using a "filter" instead of a "query" to leverage ES caching
+    query = {
+        "bool": {
+            "must": [
+                {"match": {"{}_city_name".format(scope): es_sanitize(city)}},
+                {"match": {"{}_country_code".format(scope): es_sanitize(country_code)}},
+            ]
+        }
+    }
+    if state_code:
+        # If a state was provided, include it in the filter to limit hits
+        query["bool"]["must"].append({"match": {"{}_state_code".format(scope): es_sanitize(state_code)}})
+
+    search_body = {
+        "_source": ["award_id"],
+        "size": 0,
+        "query": query,
+        "aggs": {"award_ids": {"terms": {"field": "award_id", "size": 50000}}},
+    }
+
+    return elasticsearch_results(search_body)
+
+
+def elasticsearch_results(body: dict) -> list:
```

Comment:
Function description?

Reply:
We haven't enforced function descriptions for all functions. My preferences would be to only include comments (including docstrings)  if the code is not straightforward and intuitive